### PR TITLE
BN-1124-block-packer-subgraph-score-throws-exception

### PR DESCRIPTION
**Pull request by nandotorterolo**
## Purpose
Block Packer subgraphScore throws exception

## Tickets
 closes: https://topl.atlassian.net/browse/BN-1124

### CODE REVIEW

File: minting/src/main/scala/co/topl/minting/interpreters/BlockPacker.scala [hunk 1]
```diff
@@ -285,7 +285,11 @@ object BlockPacker {
                 .toList
                 .map(_._2)
                 .foldMapM(spenders =>
-                  spenders.toList.map(_._1).map(graph.transactions).traverse(subgraphScore(graph)).map(_.max)
+                  spenders.toList
```
Comment (SeanCheatham) [inline minting/src/main/scala/co/topl/minting/interpreters/BlockPacker.scala:288]: Good catch!  I think you may be able to simplify this with an earlier match statement.  For example:

```
.foldMapM {
  case spenders if spenders.isEmpty =>
    BigInt(0).pure[F]
  case spenders =>
    spenders.toList.map(_._1) ...
}
```

**Review by SeanCheatham — APPROVED** [review]

File: minting/src/main/scala/co/topl/minting/interpreters/BlockPacker.scala [hunk 2]
```diff
@@ -285,7 +285,11 @@ object BlockPacker {
...
+                    .map(_._1)
+                    .map(graph.transactions)
+                    .traverse(subgraphScore(graph))
+                    .map(_.appended(BigInt(0)).max)
```
Comment (o-aleksandrov-topl) [inline minting/src/main/scala/co/topl/minting/interpreters/BlockPacker.scala:292]: maybe better use `maxOption` and then `getOrElse(0)`?

**Review by o-aleksandrov-topl — APPROVED** [review]

File: minting/src/main/scala/co/topl/minting/interpreters/BlockPacker.scala (hunk 1 above)
Comment (nandotorterolo) [reply minting/src/main/scala/co/topl/minting/interpreters/BlockPacker.scala:288]: thanks for the suggestion!

File: minting/src/main/scala/co/topl/minting/interpreters/BlockPacker.scala (hunk 2 above)
Comment (nandotorterolo) [reply minting/src/main/scala/co/topl/minting/interpreters/BlockPacker.scala]: thanks for the suggestion, going with Sean's review if you don't mind

**Pull request merged**